hilti/toolchain/include/compiler/detail/cxx/elements.h

```diff
@@ -106,7 +106,8 @@ class ID : public detail::IDBase<ID, normalizeID> {
     ID(std::string_view s, AlreadyNormalized n) : Base(s, n) {}
 
     /** Concatenates multiple strings into a single ID, separating them with `::`. */
-    template<typename... T, typename enable = std::enable_if_t<(... && std::is_convertible_v<T, std::string_view>)>>
+    template<typename... T,
+             typename enable = std::enable_if_t<(... && std::is_convertible_v<T, std::string_view>)>> // NOLINT
```

Comment:
Nit: Let's only silence that specific lint, and maybe insert a link to the issue. Indention gets a little funky since we don't want `NOLINT` to be part of the docstring, but that'd be fine with me.

```suggestion
    template<typename... T,
             typename enable = std::enable_if_t<
                 (... && std::is_convertible_v<
                             T, // NOLINT(modernize-type-traits), see https://github.com/llvm/llvm-project/issues/110502
                             std::string_view>)>>
```